bundles/io/org.eclipse.smarthome.io.net.security/src/main/java/org/eclipse/smarthome/io/net/security/internal/ServiceConfiguration.java

```diff
@@ -0,0 +1,265 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.io.net.security.internal;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.math.BigDecimal;
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Base64.Decoder;
+import java.util.Base64.Encoder;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * Configuration of the {@link NetworkServerTlsProvider}.
+ *
+ * @author David Graeff - Initial contribution
+ */
+public class ServiceConfiguration {
+    // Multi-context data, needs to be parsed
+    String domains = "";
```

Comment:
I understand this members are used to parse and write the configuration file more easily. Like I stated in the networkUtilities.md this is rather error prune (when editing by hand) and the parsing logic is hard to understand. I recommend reworking this part of the configuration. It should be something like the MQTT broker config in mqtt.cfg. Every broker connection has its own block of config parameters there as well.